packages/devtools_app_shared/lib/src/service/isolate_manager.dart

```diff
@@ -127,7 +134,12 @@ final class IsolateManager with DisposerMixin {
       // but that may not always be a safe assumption.
       _mainIsolate.value ??= event.isolate;
 
-      if (_selectedIsolate.value == null) {
+      if (_shouldReselectMainIsolate && event.isolate?.name == 'main') {
```

Comment:
If we're going to explicitly check for the name, we should only assign to `_mainIsolate.value` if `event.isolate?.name == 'main'` as well.

Reply:
I understand the intention but I worry about our worst case then. If our assumption about isolate naming is wrong then anything using mainIsolate will break, since the mainIsolate would always be left as null. 
With this PR's implementation the only thing that would break is isolate selection, which is a bit more forgivable.

Reply:
Maybe we should do this instead then:

```
if (_mainIsolate.value == null) {
  _mainIsolate.value = event.isolate;
  if (_shouldReselectMainIsolate) {
     // ...
  }
}
```

This way we can still only execute the `_shouldReselectMainIsolate` path if the isolate is actually what we consider to be the main isolate.